AZConnectedKubernetes fixes (#26379)
Azure/azure-powershell: src/ConnectedKubernetes/ConnectedKubernetes.Autorest/docs/Set-AzConnectedKubernetes.md

eastus   azps_test_cluster azps_test_group
```

This command enables gateway feature of a connected kubernetes cluster.

### Example 3: Enable gateway for a connected Kubernetes cluster with InputObject
```powershell
# Get an existing cluster first
$inputObject = Get-AzConnectedKubernetes -ClusterName azps_test_cluster -ResourceGroupName azps_test_group -SubscriptionId $subscriptionId
# Enable gateway and set gateway resource Id 
$outputObject.GatewayEnabled=$true
$outputObject.GatewayResourceId=$gatewayResourceId

> Set-AzConnectedKubernetes -InputObject $inputObject     

Reviewer: mix of `$outputObject` and `$inputObject`

Author: nice catch!

```

```output
Location Name              ResourceGroupName
-------- ----              -----------------
eastus   azps_test_cluster azps_test_group
```

This command enables gateway feature of a connected kubernetes cluster.

### Example 4: Enable workload identity of a connected kubernetes cluster with InputObject
```powershell
# Get an existing cluster first
$inputObject = Get-AzConnectedKubernetes -ClusterName azps_test_cluster -ResourceGroupName azps_test_group -SubscriptionId $subscriptionId
# Enable workload identity and OIDC issuer profile
$inputObject.WorkloadIdentityEnabled=$true
$inputObject.OidcIssuerProfileEnabled=$true
Set-AzConnectedKubernetes -InputObject $inputObject 
```

```output
Location Name              ResourceGroupName
-------- ----              -----------------
eastus   azps_test_cluster azps_test_group
```

> This command enables workload identiy and Oidc Issuer Profile of a connected kubernetes cluster

Reviewer: ```suggestion
This command enables workload identity and OIDC Issuer Profile for a connected Kubernetes cluster
```

### Example 5: Disable workload identity of a connected kubernetes cluster with InputOjbect
```powershell
# Get an existing cluster first
$inputObject = Get-AzConnectedKubernetes -ClusterName azps_test_cluster -ResourceGroupName azps_test_group -SubscriptionId $subscriptionId
# Disable workload identity 
$inputObject.WorkloadIdentityEnabled=$falue
Set-AzConnectedKubernetes -InputObject $inputObject 
```

```output
Location Name              ResourceGroupName
-------- ----              -----------------
eastus   azps_test_cluster azps_test_group
```

This command disables workload identity of a connected kubernetes cluster

### Example 6: Disable workload identity of a connected kubernetes cluster
```powershell
Get-AzConnectedKubernetes -ClusterName azps_test_cluster -ResourceGroupName azps_test_group -SubscriptionId $subscriptionId | Set-AzConnectedKubernetes -WorkloadIdentityEnabled:$false
```

```output
Location Name              ResourceGroupName
-------- ----              -----------------
eastus   azps_test_cluster azps_test_group
```

> This command disables workload identiy of a connected kubernetes cluster

Reviewer: ```suggestion
This command disables workload identity for a connected Kubernetes cluster
```

> ### Example 2: Enable gateway feature of connected kubernetes.

Reviewer: ```suggestion
### Example 2: Enable gateway for a connected Kubernetes cluster
```

Think writing it like this matches our public docs better

(here and elsewhere)

eastus   azps_test_cluster azps_test_group
```

This command enables gateway feature of a connected kubernetes.

### Example 3: Enable gateway feature of connected kubernetes with InputObject
```powershell

> # Get an existing clustser first

Reviewer: ```suggestion
# Get an existing cluster first
```

$inputObject = Get-AzConnectedKubernetes -ClusterName azps_test_cluster -ResourceGroupName azps_test_group -SubscriptionId $subscriptionId
# Enable gateway and set gateway resource Id 
$outputObject.GatewayEnabled=$true
$outputObject.GatewayResourceId=$gatewayResourceId

> Set-AzConnectedKubernetes -InputObject $inputObject     

Reviewer: Wouldn't you do this by making a call like this?

```PowerShell
Get-AzConnectedKubernetes -ClusterName azps_test_cluster -ResourceGroupName azps_test_group -SubscriptionId $subscriptionId
| Set-AzConnectedKubernetes -GatewayResourceId $gatewayResourceId
```

Reviewer: (similar applies to other examples below - they all look fine, but we introduced `-WorkloadIdentityDisabled` to make this possible)
